server/src/main/java/com/microsoft/java/bs/core/internal/services/BuildTargetService.java

```diff
@@ -329,14 +331,21 @@ public CompileResult compile(CompileParams params) {
     if (params.getTargets().isEmpty()) {
       return new CompileResult(StatusCode.OK);
     } else {
-      StatusCode code = runTasks(params.getTargets(), this::getBuildTaskName);
+      Function<Map<String, Set<BuildTargetIdentifier>>,
+          Function<BuildClient, ProgressReporter>> reporterCreator =
+              fullTaskPathMap -> buildClient ->
+                  new CompileProgressReporter(client, params.getOriginId(), fullTaskPathMap);
+      StatusCode code = runTasks(params.getTargets(), this::getBuildTaskName, reporterCreator);
       CompileResult result = new CompileResult(code);
       result.setOriginId(params.getOriginId());
 
       // Schedule a task to refetch the build targets after compilation, this is to
       // auto detect the source roots changes for those code generation framework,
       // such as Protocol Buffer.
-      CompletableFuture.runAsync(this::reloadWorkspace);
+      // Ignore test origin id as it causes more messages to be sent.
+      if (!"do not reload workspace".equals(params.getOriginId())) {
```

Comment:
Where does the `do not reload workspace` comes from?

Reply:
The integrity tests check the start/finish message count for a compilation to ensure they line up with the number of build targets.  `CompletableFuture.runAsync(this::reloadWorkspace)` triggers additional start/finish messages which I don't want to see for these tests.
The use of `OriginId` was a cheat.  I'll change this to a system property e.g. `Boolean.getBoolean("bsp.plugin.reloadworkspace.enabled")` and have `bsp.plugin.reloadworkspace.enabled=false` in this case.

Reply:
Thank you, system property looks better :)